### DISCUSSION TITLE
Qt 6.2 support

### DISCUSSION THREAD

**Discussion by madcodingrocks** [MeVisLab/pythonqt#77] [Q&A]
Hi, sorry for asking, but I did not find any related information around.
Is there any support available or planned for Qt 6.2?

I tried to build the generated files with the generator with not much success. is there any doc about how that command works?

thanks in advance!

**Comment by mrbean-bremen** [comment]
There is no support yet, see also #13. As far as I know, this is planned for some time next year.

**Reply by madcodingrocks** [reply]
thanks for the prompt reply! very much appreciated

**Reply by Lbardoux** [reply]
Hello @mrbean-bremen, supporting Qt6.2 is a great news. My company will really need the support for Qt6.2 for 2023, do you have more information about when it is expected to be delivered ?
In the mean time, as a backup plan, I have started a local branch where I just adapt the existing generated_cpp_5 to compile and target qt6.2 (plus adjustements in PythonQt code to compile/get rid of the bugs Qt6 has created). I'm not touching the generator for now

Do you think my approach is a good idea?

**Reply by mrbean-bremen** [reply]
I don't have a timeframe for this yet, accept that this will be later this year. We probably will go with replacing the generator by [Shiboken](https://pypi.org/project/shiboken6/) , as has been suggested by @florianlink.
As for directly adapting the generated code: if that works for you, sure - though it would not make much sense to merge this back. If you want to have a go at Shiboken, even better 😁

**Reply by Lbardoux** [reply]
Hello @mrbean-bremen, Yes indeed, the modified generated code shall not be merged. What about the code changes to PythonQt.cpp? There is quite few already to support Qt6, like changing the aliases for QVector and QPair (which now are respectively typedef over QList and std::pair) or the new behavior of QVariant::operator= (it was no longer possible to call a slot taking a QString/QStringList as argument because of this new behavior)

> I'll be glad to participate to the changes for the generator code if it is possible

Best regards

**Reply by mrbean-bremen** [reply]
> What about the code changes to PythonQt.cpp?

As long as they don't break compatibility with older version (which hopefully would be caught by the CI checks), that should be fine, thanks!
